infra-cdk/lib/stacks/IAMStack.ts

```diff
@@ -3,11 +3,14 @@ import * as iam from 'aws-cdk-lib/aws-iam';
 import { Construct } from 'constructs';
 import { ENVIRONMENT } from '../core/constants';
 import { BaseStackProps } from '../core/types';
+import { S3BucketStack } from '../stacks/ReportsBucketStack';
 
 export type IAMStackProps = {
     /** DynamoDB table name to provide permissions to */
-    tableName: string;
-    reportsTableName: string;
+    tableName?: string;
+    reportsTableName?: string;
```

Comment:
Is there a reason to make these optional? Won't we always want the application role to have access to a messages table, a reports table, and a bucket? Making them mandatory would remove a lot of unnecessary edge cases from this code :)